src/lib/builders/foreachstate-builder.ts

```diff
@@ -28,6 +28,10 @@ import { validators } from '../validators';
 function foreachstateBuildingFn(data: Specification.Foreachstate): () => Specification.Foreachstate {
   return () => {
     data.type = 'foreach';
+
+    //FIXME https://github.com/serverlessworkflow/sdk-typescript/issues/95
+
+    data.usedForCompensation = false;
```

Comment:
IMO this is wrong on 2 levels:

- Codewise, this **always** sets `usedForCompensation` to `false`. So even if someone intentionally sets it to `true`, it will be overridden.
- Schemawise this is not what the JSON Schema says so it's an opiniated choice to "default" this value to `false`.

Reply:
I take back what I said about the schema, it does indeed specify a default.

The proper way to check would then be:
```typescript
data.usedForCompensation = data.usedForCompensation || false;
```